### DISCUSSION TITLE
Cache Gatsby GraphQL Schema To Reduce Build Time?

### DISCUSSION THREAD

**Discussion by jasonbiondo** [gatsbyjs/gatsby#27596] [Help] [answered]
Is it possible to just cache the entire schema of all of the data to reduce the time it takes for the future builds on development?  It takes a lot of time to do the node creations and having to restart dev for minutes on end is unproductive for my team.  I'm looking for a solution on this. Does anyone know how this can be done?

**Comment by KyleAMathews** [comment]
Yup! https://www.gatsbyjs.com/plugins/gatsby-plugin-schema-snapshot/

**Comment by jasonbiondo** [comment, marked as answer]
This does indeed grab the schema but it doesn't cache the data.  I don't want to rerun the same plugins over and over.  Is there a way to cache all the data and then load it in?

**Reply by wardpeet** [reply]
At this moment no, sourcing will always need to download assets from the source. You could modify the source plugin to bake in a zip cache or something.

We're planning to improve the speed of the develop server but it won't fix sourcing. See https://github.com/gatsbyjs/gatsby/discussions/27603

**Reply by KyleAMathews** [reply]
Individual source plugins can (and many do) cache data locally e.g. gatsby-source-contentful & gatsby-source-wordpress — what source plugin(s) are you using?

https://www.gatsbyjs.com/docs/creating-a-source-plugin/#improve-plugin-developer-experience-by-enabling-faster-sync
